libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/MultiLanguageGeneratorBase.cs

```diff
@@ -40,20 +42,34 @@ public MultiLanguageGeneratorBase()
         /// <returns>The generator.</returns>
         public override async Task<string> Generate(ITurnContext turnContext, string template, object data)
         {
-            // see if we have any locales that match
             var targetLocale = turnContext.Activity.Locale?.ToLower() ?? string.Empty;
 
-            var locales = new string[] { string.Empty };
-            if (!this.LanguagePolicy.TryGetValue(targetLocale, out locales))
+            // priority 
+            // 1. local policy
+            // 2. shared policy in turnContext
+            // 3. default policy
+            var languagePolicy = this.LanguagePolicy ?? 
+                                turnContext.TurnState.Get<LanguagePolicy>() ?? 
+                                new LanguagePolicy();
+
+            // see if we have any locales that match, if not, use empty as fallback
+            var fallbackLocales = new List<string>();
+            if (languagePolicy.ContainsKey(targetLocale))
             {
-                if (!this.LanguagePolicy.TryGetValue(string.Empty, out locales))
-                {
-                    throw new Exception($"No supported language found for {targetLocale}");
-                }
+                fallbackLocales.AddRange(languagePolicy[targetLocale]);
+            }
+            else
+            {
+                fallbackLocales.AddRange(languagePolicy[string.Empty]);
```

Comment:
> fallbackLocales.AddRange(languagePolicy[string.Empty]); [](start = 15, length = 56)

this will throw if there is no "empty" entry .  I think I prefer the TryGetvalue approach better.